### ISSUE TITLE
ns.exec() command not working for Weaken and Hack; but does for Grow

### ISSUE THREAD

**Issue by afsanchez001** [afsanchez001/BitBurner#1]
NOTE: This is an OPEN QUESTION to The Black Hat, and I have made this public to share with others who may have experienced the same error. Applies to tutorial @ https://www.youtube.com/watch?v=nePsChf_Ifk 

THE PROBLEM: "ns.exec not working for two of the implementations in Go.js"

In Go.js there are three uses of the ns.exec() command.

ns.exec("bin.wk.js", server, available_threads, target); // Weaken the target while security > minSecurity.

ns.exec("bin.gr.js", server, available_threads, target); // Grow the target while money < maxMoney.

ns.exec("bin.hk.js", server, available_threads); // Hack the target

weaken and hack both produce this error: 'hostname' should be a string. Is undefined.

But grow runs and does not produce that error. 

I am using v2.1.0 (8f4636cb)

**Comment by afsanchez001** [comment]
Looking at the exec() Netscript Function @ https://bitburner.readthedocs.io/en/latest/netscript/basicfunctions/exec.html

The usage shows this:

exec("generic-hack.script", "foodnstuff");

But in the Go.js, the target is not encapsulated with quotes. Rather the term server is sent as an 'any'.

I tried sending it as server.hostname (which is a String), but that did not work.

The only instance working with just the server parameter as 'any' is ns.grow(server); the others, ns.weaken(server) and ns.hack(server) fail with the error "hostname should be a string. Is undefined"

**Comment by afsanchez001** [comment]
Added this line in Go.js 'main' to see what I was getting from the server list array.

ns.print("Analyzing server, " + server.hostname); // 10-30-2022, NEW for debugging.

/*
So, the above shows as Undefined in Tail Go.js window.
			
This means that the value of the server is not there. But why?
			
I am going to focus on the function serverList() aka dpList() in the tutorial.

I did see a comment in the Video comments area by user Kyle Walters that the script is not iterating through the servers properly.

Related? Not sure.

*/
